types/selenium-webdriver/tsconfig.json

```diff
@@ -5,6 +5,7 @@
         "lib": [
             "es6"
         ],
+        "esModuleInterop": true,
```

Comment:
This option should not be added to the config.

Reply:
@jakebailey - Sorry for the delay, done

Reply:
@jakebailey - Aaand now the CI is failing. Got any tips for how to correctly import this thing?

https://github.com/toddtarsi/DefinitelyTyped/blob/patch-2/types/selenium-webdriver/index.d.ts#L19

Reply:
Is it not just like every other import in the file?

Reply:
@jakebailey - No, I believe this file exports this function as `module.exports = fn`

Reply:
Sure, then the test file still has the old form:

```
import LogInspector from "selenium-webdriver/bidi/logInspector";
```

But, I'm not totally sure I understand why this package is using mixed import styles. That seems wrong, one way or another.

Reply:
Yeah, I agree

Reply:
I believe this got me through the linter. This codebase has some very strict requirements on how code hs to be declared.

Reply:
It had to be strict; without tsc generating both the js and the d.ts, there's no guarantee that things are correct.

Reply:
@jakebailey - If you'd like to throw a commit on this PR, knock yourself out. I'm not really sure what to do here for you anymore.

Reply:
I'm confused, I'm trying to help you fix the lint... is there code unpushed that you want me to review? Something else?

Reply:
The issue is now because the import, nothing linter related. esModuleInterop was what got that import working. The linting is unrelated.

```
Module '"/home/runner/work/DefinitelyTyped/DefinitelyTyped/types/selenium-webdriver/bidi/logInspector"' can only be default-imported using the 'esModuleInterop' flag
```

It's like I had a passing PR, passing CI, passing linter, and then you came in with config changes that were needed because of the underlying package.

Reply:
The CI isn't strict about this one flag, but the bot comment at the start of the PR notes it as nearly every package should not be setting this flag and so it's up to a maintainer to double check that it's needed. Given this is an existing package, follow-up changes almost never need to change the flag. See: https://github.com/DefinitelyTyped/DefinitelyTyped#tsconfigjson

I'm sorry that you had a green CI and the change felt complete, but there are still human steps here and that's why this PR was flagged for maintainer review and why I'm reviewing it.

Reply:
Yeah, I see that you were called by the check config stuff. I'm sorry to be an ass at you. You hit my frustrations dead on. I'm sorry to yell at you when you're using your Saturday to help me.

It's needed in this case. I wish they had used a key on the export as well, but unfortunately they didn't. I'll look into changing that import to be normal, but unfortunately, I think doing a normal default import might require adjusting linter rules too

Reply:
@jakebailey - Wait, did you avoid using esModuleInterop by just using that syntax to import?

Reply:
`esModuleInterop` is special and instructs TS to emit special helpers to help convert between `export =` and `export default`, both at the _export_ site and the _import_ site. Since DT describes modules that _aren't_ written in TS, we always leave `esModuleInterop` disabled to force DT to "do the right thing" and describe the code exactly as it was written upstream without special helpers, such that consumers of types packages always have accurate info about what they're importing.

I just applied the change I suggested to the tests; you had previously only applied it to the exported types and that's what I was trying to say to fix in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66873#discussion_r1359111753.

Of course, I'm 99% certain that all of the other files are actually also wrong; reading the unpkg page for this package shows that all of the modules are using `module.exports =` (would be `export =` in types) but not in the DT package! But your PR does the right thing, which _looks_ weird becuase it doesn't match anything else, but you're not making the problem worse, so that's good.

Reply:
😅 Thank you for explaining that. I uh, I have all the pieces to understand that, but it may take some time to sink in

Reply:
Believe me when I say I don't fully understand it myself!